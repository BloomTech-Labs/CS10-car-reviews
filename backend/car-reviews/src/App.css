.App {
  text-align: center;
  font-family: 'Montserrat', sans-serif;
  border: none !important;
}

h3 {
  font-family: 'Julius Sans One', sans-serif;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}
/*.example{
  border: 1px solid red; 
  display: flex !important;
  justify-content: center;
}*/

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
