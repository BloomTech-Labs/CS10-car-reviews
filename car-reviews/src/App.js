import React, { Component } from 'react';
import './App.css';
import MainPage from './components/MainPage/mainpage';
import MyReviews from './components/UserPage/myreviews';
import { Switch, Route, Redirect } from 'react-router-dom';
import BillingContainer from './components/UserPage/billingcontainer';
import Settings from './components/UserPage/settings';
import SearchResults from './components/MainPage/searchresults';
import Login from './components/MainPage/loginregister';
import AuthService from './components/Auth/authservice';

const Auth = new AuthService();

const PrivateRoute = ({ component: Component, ...rest }) => (
  <Route
    {...rest}
    render={props =>
      Auth.loggedIn() === true ? <Component {...props} /> : <Redirect to="/login" />
    }
  />
);

class App extends Component {
  render() {
    return (
      <div className="App">
        <Switch>
          <Route exact path="/" component={MainPage} />
          <Route path="/searchpage" component={SearchResults} />
          <PrivateRoute path="/Billing" component={BillingContainer} />
          <PrivateRoute path="/MyReviews" component={MyReviews} />
          <PrivateRoute path="/Settings" component={Settings} />
          <Route path="/Login" component={Login} />
        </Switch>
      </div>
    );
  }
}

export default App;
