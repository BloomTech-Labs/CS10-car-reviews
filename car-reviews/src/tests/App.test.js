import App from '../App';

describe('<App />', () => {
  it('renders App component correctly', () => {
    const app = shallow(<App />);
    expect(app).toMatchSnapshot();
  });

  it('always renders a div', () => {
    const divs = shallow(<App />).find('div');
    expect(divs.length).toBe(1);
  });

  describe('the rendered div', () => {
    it('contains everything else that gets rendered', () => {
      const divs = shallow(<App />).find('div');
      const wrappingDiv = divs.first();

      // Enzyme omits the outermost node when using the .children() method on shallow(<App/>).
      expect(wrappingDiv.children()).toEqual(shallow(<App />).children());
    });
  });

  describe('Route component', () => {
    it('should render exactly three <Route/> components', () => {
      const app = shallow(<App />);
      const route = app.find('Route');
      expect(route.length).toEqual(3);
    });

    it("First Route component's exact path is '/' and receives three props", () => {
      const route = shallow(<App />).find('Route');
      expect(route.at(0).props().path).toEqual('/');
      expect(route.at(0).props().exact).toBeTruthy();
      expect(Object.keys(route.at(0).props()).length).toBe(3);
    });

    it("Second Route component's non-exact path is '/searchpage' and receives two props", () => {
      const route = shallow(<App />).find('Route');
      expect(route.at(1).props().path).toEqual('/searchpage');
      expect(route.at(1).props().exact).toBeFalsy();
      expect(Object.keys(route.at(1).props()).length).toBe(2);
    });

    it("Third Route component's non-exact path is '/Login' and receives two props", () => {
      const route = shallow(<App />).find('Route');
      expect(route.at(2).props().path).toEqual('/Login');
      expect(route.at(2).props().exact).toBeFalsy();
      expect(Object.keys(route.at(2).props()).length).toBe(2);
    });
  });

  describe('PrivateRoute component', () => {
    it('should render exactly three <PrivateRoute/> components', () => {
      const app = shallow(<App />);
      const privateroute = app.find('PrivateRoute');
      expect(privateroute.length).toEqual(3);
    });

    it("First PrivateRoute component's non-exact path is '/Billing' and receives two props", () => {
      const privateroute = shallow(<App />).find('PrivateRoute');
      expect(privateroute.at(0).props().path).toEqual('/Billing');
      expect(privateroute.at(0).props().exact).toBeFalsy();
      expect(Object.keys(privateroute.at(0).props()).length).toBe(2);
    });

    it("Second PrivateRoute component's non-exact path is '/MyReviews' and receives two props", () => {
      const privateroute = shallow(<App />).find('PrivateRoute');
      expect(privateroute.at(1).props().path).toEqual('/MyReviews');
      expect(privateroute.at(1).props().exact).toBeFalsy();
      expect(Object.keys(privateroute.at(1).props()).length).toBe(2);
    });

    it("Third PrivateRoute component's non-exact path is '/Settings' and receives two props", () => {
      const privateroute = shallow(<App />).find('PrivateRoute');
      expect(privateroute.at(2).props().path).toEqual('/Settings');
      expect(privateroute.at(2).props().exact).toBeFalsy();
      expect(Object.keys(privateroute.at(2).props()).length).toBe(2);
    });
  });
});
