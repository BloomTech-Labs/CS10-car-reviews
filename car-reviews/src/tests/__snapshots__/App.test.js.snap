// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders App component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
                component={[Function]}
                exact={true}
                path="/"
        />
        <Route
                component={[Function]}
                path="/searchpage"
        />
        <PrivateRoute
                component={[Function]}
                path="/Billing"
        />
        <PrivateRoute
                component={[Function]}
                path="/MyReviews"
        />
        <PrivateRoute
                component={[Function]}
                path="/Settings"
        />
        <Route
                component={[Function]}
                path="/Login"
        />
</Switch>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            path="/searchpage"
/>,
          <PrivateRoute
            component={[Function]}
            path="/Billing"
/>,
          <PrivateRoute
            component={[Function]}
            path="/MyReviews"
/>,
          <PrivateRoute
            component={[Function]}
            path="/Settings"
/>,
          <Route
            component={[Function]}
            path="/Login"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/searchpage",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "path": "/Billing",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "path": "/MyReviews",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "path": "/Settings",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/Login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    path="/searchpage"
          />
          <PrivateRoute
                    component={[Function]}
                    path="/Billing"
          />
          <PrivateRoute
                    component={[Function]}
                    path="/MyReviews"
          />
          <PrivateRoute
                    component={[Function]}
                    path="/Settings"
          />
          <Route
                    component={[Function]}
                    path="/Login"
          />
</Switch>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              path="/searchpage"
/>,
            <PrivateRoute
              component={[Function]}
              path="/Billing"
/>,
            <PrivateRoute
              component={[Function]}
              path="/MyReviews"
/>,
            <PrivateRoute
              component={[Function]}
              path="/Settings"
/>,
            <Route
              component={[Function]}
              path="/Login"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/searchpage",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "component": [Function],
              "path": "/Billing",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "component": [Function],
              "path": "/MyReviews",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "component": [Function],
              "path": "/Settings",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/Login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
